test/qa/test_meshnet.py

```diff
@@ -88,31 +89,211 @@ def test_mesh_removed_machine_by_other():
-        peer_hostname = meshnet.get_this_device(ssh_client.exec_command("nordvpn mesh peer list"))
-        # Initiate ssh connection via mesh because we are going to lose the main connection
-        ssh_client_mesh = ssh.Ssh(peer_hostname, "root", "root")
-        ssh_client_mesh.connect()
-        with lib.Defer(ssh_client_mesh.disconnect):
-            ssh_client_mesh.exec_command("nordvpn c")
-            with lib.Defer(lambda: ssh_client_mesh.exec_command("nordvpn d")):
-                # We should not have direct connection anymore after connecting to VPN
-                with pytest.raises(sh.ErrorReturnCode_1) as ex:
-                    assert "icmp_seq=" not in sh.ping("-c", "1", "qa-peer")
-
-                    ssh_client_mesh.exec_command(f"nordvpn allowlist add subnet {my_ip}/32")
-                    with lib.Defer(lambda: ssh_client_mesh.exec_command("nordvpn allowlist remove all")):
-                        # Direct connection should work again after allowlisting
-                        assert "icmp_seq=" in sh.ping("-c", "1", "qa-peer")
+    my_ip = ssh_client.exec_command("echo $SSH_CLIENT").split()[0]
+
+    peer_hostname = meshnet.get_this_device(ssh_client.exec_command("nordvpn mesh peer list"))
+    # Initiate ssh connection via mesh because we are going to lose the main connection
+    ssh_client_mesh = ssh.Ssh(peer_hostname, "root", "root")
+    ssh_client_mesh.connect()
+    with lib.Defer(ssh_client_mesh.disconnect):
+        ssh_client_mesh.exec_command("nordvpn c")
+        with lib.Defer(lambda: ssh_client_mesh.exec_command("nordvpn d")):
+            # We should not have direct connection anymore after connecting to VPN
+            with pytest.raises(sh.ErrorReturnCode_1) as ex:
+                assert "icmp_seq=" not in sh.ping("-c", "1", "qa-peer")
+
+                ssh_client_mesh.exec_command(f"nordvpn allowlist add subnet {my_ip}/32")
+                with lib.Defer(lambda: ssh_client_mesh.exec_command("nordvpn allowlist remove all")):
+                    # Direct connection should work again after allowlisting
+                    assert "icmp_seq=" in sh.ping("-c", "1", "qa-peer")
+
+
+def validate_input_chain(peer_ip: str, routing: bool, local: bool, incoming: bool, fileshare: bool) -> (bool, str):
+    rules = sh.sudo.iptables("-S", "INPUT")
+
+    fileshare_rule = f"-A INPUT -s {peer_ip}/32 -p tcp -m tcp --dport 49111 -m comment --comment nordvpn -j ACCEPT"
+    if (fileshare_rule in rules) != fileshare:
+        return (False, f"Fileshare permissions configured incorrectly, rule expected: {fileshare}\nrules:{rules}")
+
+    incoming_rule = f"-A INPUT -s {peer_ip}/32 -m comment --comment nordvpn -j ACCEPT"
+    if (incoming_rule in rules) != incoming:
+        return (False, f"Incoming permissions configured incorrectly, rule expected: {incoming}\nrules:{rules}")
+
+    incomig_rule_idx = rules.find(incoming_rule)
+
+    if not incoming_rule:
+        return (True, "")
+
+    for lan in meshnet.LANS:
```

Comment:
What rules are being checked here? I don't remember any meshnet related lan rules in the input chain, only in forward chain.